cfme/tests/networks/test_sdn_inventory_collection.py

```diff
@@ -125,16 +129,18 @@ def test_sdn_nsg_firewall_rules(provider, appliance, secgroup_with_rule):
     """
 
     # Navigate to network provider.
-    collection = appliance.collections.network_providers.filter({'provider': provider})
-    network_provider = collection.all()[0]
+    prov_collection = appliance.collections.network_providers.filter({'provider': provider})
+    network_provider = prov_collection.all()[0]
+    network_provider.refresh_provider_relationships()
+    wait_for(network_provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600)
     view = navigate_to(network_provider, 'Details')
     parent_name = view.entities.relationships.get_text_of("Parent Cloud Provider")
     assert parent_name == provider.name
 
-    # Navigate to secgroup.
-    collection = appliance.collections.network_security_groups.filter({'name': secgroup_with_rule})
-    secgroup = collection.all()[0]
+    secgrp_collection = appliance.collections.network_security_groups.filter(
+        {'name': secgroup_with_rule})
+    secgroup = secgrp_collection.all()[0]
     view = navigate_to(secgroup, 'Details')
 
-    assert 'Port' == view.entities.firewall_rules[0][3].text
-    assert '22' == view.entities.firewall_rules[1][3].text
+    assert 'Port' == view.entities.firewall_rules[1][3].text
```

Comment:
Why the change here to which row is indexed? There needs to be an in-line comment explaining these 'magic numbers' at the minimum, so we're not guessing at what row _should_ have been there.